packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import TableCaption from './TableCaption';
+import classNames from 'classnames';
+
+function debounce(fn, ms) {
+  let timer;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      timer = null;
+      fn.apply(this, arguments);
+    }, ms);
+  };
+}
+
+export const Table = ({
+  className,
+  stackBreakpoint,
+  striped,
+  scrollable,
+  children,
+  ...tableProps
+}) => {
+  const container = useRef(null);
+  // The captionID is calculated and stored as init value of a ref.
+  // This ensures that the ID remains constant for all renders.
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
+  const [isTableScrollable, setTableScrollable] = useState(false);
+
+  // Listens to the window resize event to dynamically handle horizontal scrollable tables.
+  useEffect(() => {
+    // Set isTableScrollable `true` if the table width is wider than the viewport and Table is set for horizontal scrolling.
+    if (scrollable) {
+      const { scrollWidth, clientWidth } = container.current;
+      const isScrollActive = scrollWidth > clientWidth;
+      setTableScrollable(isScrollActive);
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { scrollWidth, clientWidth } = container.current;
+        const isScrollActive = scrollWidth > clientWidth;
+        setTableScrollable(isScrollActive);
+      }, 500);
+
+      // Create window resize event listener that calls the debouncedHandleResize function
+      window.addEventListener('resize', debouncedHandleResize);
+
+      // Clean up the establishd event listeners
+      return () => {
+        window.removeEventListener('resize', debouncedHandleResize);
+      };
+    }
+  });
```

Comment:
Relaying feedback from chat:

https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
theres no need to create and destroy the window listeners each time if we pass in an extra parameter so the react hook can tell when its actually necessary to update the window listeners. I would probably use the scrollable prop as the parameter to check